### PR TITLE
contrib/pnio: Add Identification and Maintenance (I&M) blocks

### PR DESCRIPTION
**Pull request by cpackham-atlnz**
Add definitions and tests for I&M 0-4. I&M5 is more complicated so that
hasn't been included for now.

This PR touches the same files as https://github.com/secdev/scapy/pull/3296 If there is a conflict it should be trivial.

### CODE REVIEW

File: test/contrib/pnio_rpc.uts [hunk 1]
```diff
@@ -204,6 +204,30 @@ p == IODWriteMultipleRes(ARUUID='01234567-89ab-cdef-0123-456789abcdef', recordDa
     ]) / conf.padding_layer(b'\xef')
 
 
+####################################################################
+####################################################################
+
++ Check I&M
+
+= IM0Block default values
+bytes(IM0Block()) == bytearray.fromhex('002000380100000000000000000000000000000000000000000000000000000000000000000000000000000000005600000000000000000001010000')
```
Comment (guedou) [inline test/contrib/pnio_rpc.uts:213]: Call you use `raw()` instead of `bytes()` in your tests?

**Review by guedou — DISMISSED** [review]

File: test/contrib/pnio_rpc.uts (hunk 1 above)
Comment (cpackham-atlnz) [reply test/contrib/pnio_rpc.uts:213]: I can but the rest of the tests in this file use `bytes()`. Should I do a commit to convert the others before adding my new changes as a follow up?

Comment (guedou) [reply test/contrib/pnio_rpc.uts]: No. This is a fine as it is. Thanks for the fix!

File: scapy/contrib/pnio_rpc.py [hunk 2]
```diff
@@ -626,6 +626,77 @@ def post_build(self, p, pay):
...
+    block_type = 0x0022
+
+
+#     I&M3
+class IM3Block(Block):
+    """Identification and Maintenance 3"""
+    fields_desc = [
+        BlockHeader,
+        StrFixedLenField("IMDescriptor", "", length=54),
+
```
Comment (gpotter2) [inline scapy/contrib/pnio_rpc.py:683]: Useless space

**Review by gpotter2 — COMMENTED** [review]
LGTM but TBH I'm quite surprised this passes flake8 tests

File: scapy/contrib/pnio_rpc.py (hunk 2 above)
Comment (cpackham) [reply scapy/contrib/pnio_rpc.py:683]: probably get away with because it's between square brackets. Will rebase and fix

Comment (cpackham-atlnz) [reply scapy/contrib/pnio_rpc.py]: Done. I think flake8 allowed it because it was between square brackets

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**